src/AppBanner/index.tsx

```diff
@@ -138,28 +115,75 @@ export const AppBanner = (props: Props): ReactElement | null => {
                         </Button>
                     </ThemeProvider>
                 </div>
-                <div css={commonStyles.bottomRightComponent}>
-                    <div css={styles.image}>
+                <div css={styles.bottomRightComponent}>
+                    <div css={localStyles.image}>
                         <img src={imageUrl} alt="" />
                     </div>
-                    <div css={commonStyles.iconPanel}>
-                        <ThemeProvider theme={overrridenReaderRevenueTheme}>
-                            <Button
-                                icon={<SvgCross />}
-                                hideLabel={true}
-                                css={commonStyles.closeButton}
-                                priority="tertiary"
-                                size="small"
-                                aria-label="Close"
-                                onClick={(e) => onCloseClick(e, CLOSE_BUTTON_ID)}
-                                tabIndex={0}
-                            >
-                                {' '}
-                            </Button>
-                        </ThemeProvider>
-                    </div>
+                    <BannerCloseButton
+                        trackClick={trackClick}
+                        setShowBanner={setShowBanner}
```

Comment:
Copied from #459:

> What do you think about simplifying the interface to BannerCloseButton and passing through a single callback to invoke when the button is clicked? I think that could be nicer as there's less coupling between the concerns of this component and more generic BannerCloseButton.

To clarify, what if we passed through a callback which was invoked by `BannerCloseButton` when clicked which did both click tracking and toggling the banner visibility? It could have the same signature as the `onCloseAction` method in `BannerCloseButton`. So the button ID is supplied by `BannerCloseButton` (something it knows about) but toggling visibility and `trackClick` are something the parent knows about, so the detail isn't leaked through to `BannerCloseButton`.

Reply:
Sorry - I'm not sure I follow. Closing the banner, and tracking anything in the banner that the user has interacted with, are different things. I'm not keen on combining them in a single callback - which I assume would have to be hardcoded into every banner component?

Reply:
Sorry I didn't explain well. You're right that they're separate things, but as far as the close button component is concerned maybe it's better to keep it "dumb" - I think all it should know about is "I've got a callback to execute when I get clicked". The callback is provided by the parent component which _does_ know about setting state and tracking etc but the child component can be blissfully ignorant, all it knows is I've got to execute something when I'm clicked.

Reply:
Ah. I see where you're coming from. The way I've built the close button component ... it's not living in blissful ignorance: it knows that its purpose is to close a banner that includes it. It handles both closing the banner (via clicking on it, and by user pressing the ESC key), and reporting that user has interacted with it. It's the banner that doesn't care about tracking the close action, because it includes the close button component which handles all that for it.

The App banner component is the only banner which has to be different, in that it will also close when the user clicks on either of its CTA buttons. Hence the need to duplicate the onCloseClick and onCloseAction functions in this banner's code. 

Does that make sense?